src/audio/dai.c

```diff
@@ -548,12 +548,151 @@ static int dai_params(struct comp_dev *dev,
 		dai_capture_params(dev, period_bytes, period_count);
 }
 
+static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
+{
+	struct sof_ipc_comp_config *dconfig = dev_comp_config(dev);
+	struct dai_data *dd = comp_get_drvdata(dev);
+	struct sof_ipc_comp_dai *dai = COMP_GET_IPC(dev, sof_ipc_comp_dai);
+	int channel = 0;
+	int handshake;
+	int ret = 0;
+
+	comp_info(dev, "dai_config() dai type = %d index = %d dd %p",
+		  config->type, config->dai_index, dd);
+
+	/* cannot configure DAI while active */
+	if (dev->state == COMP_STATE_ACTIVE) {
+		comp_info(dev, "dai_config(): Component is in active state. Ignore config");
+		return 0;
+	}
+
+	if (dd->chan) {
+		comp_info(dev, "dai_config(): Configured. dma channel index %d, ignore...",
+			  dd->chan->index);
+		return 0;
+	}
+
+	if (config->group_id) {
+		ret = dai_assign_group(dev, config->group_id);
+
+		if (ret)
+			return ret;
+	}
+
+	switch (config->type) {
+	case SOF_DAI_INTEL_SSP:
```

Comment:
Can we have an op here that dai.c can call here. This way we can decouple dai.c from DAI specific logic.

Reply:
> Can we have an op here that dai.c can call here. This way we can decouple dai.c from DAI specific logic.

Good idea, just gather it in a new created helper dai_config_dma_channel(), may change it to DAI op later if needed.